Update @wordpress/components to the latest version

vbelolapotkov opened the pull request:
Fixes #717

#### Changes proposed in this Pull Request

* Update `@wordpress/components` dependency
* Fixes all UI quirks after update
* Resolves instant-deposits UI TODOs introduced by #1437

#### Testing instructions

* Checks should be green.
* Update dependencies with `npm install`.
* Launch site and check the UI.

-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

vbelolapotkov commented on package.json:
```diff
@@ -135,7 +136,7 @@
     "ts-loader": "8.0.17",
     "typescript": "4.1.5",
-    "webpack": "5.24.1",
+    "webpack": "^4.46.0",
```
Note for reviewer. Components doesn't work well with webpack 5, so reverted to v4. More context in https://github.com/WordPress/gutenberg/issues/27984#issuecomment-754530982

vbelolapotkov replied:
Decision was discussed in this thread: p1618406589275200-slack-CGGCLBN58

vbelolapotkov commented on webpack.config.js:
```diff
@@ -54,12 +54,12 @@ const webpackConfig = {
-				exclude: /node_modules/,
-				loader: 'source-map-loader',
-			},
+			// {
+			// 	enforce: 'pre',
+			// 	test: /\.js$/,
+			// 	exclude: /node_modules/,
+			// 	loader: 'source-map-loader',
+			// },
```
@reykjalin need you help with this piece. I had to downgrade webpack to v4 (https://github.com/Automattic/woocommerce-payments/pull/1536/files#r610672441) and this config section was causing errors. I see it was introduced [here](https://github.com/Automattic/woocommerce-payments/pull/1263/files#diff-1fb26bc12ac780c7ad7325730ed09fc4c2c3d757c276c3dacc44bfe20faf166f) to enable TS support. 
Is it required because of webpack 5 or for TS in general? Do you know any workaround solution for TS setup which would work with webpack v4?

vbelolapotkov replied:
Resolved in https://github.com/Automattic/woocommerce-payments/pull/1536/commits/5d83fbbe46bfdb001b387e23c111156abc3d81bc

vbelolapotkov replied:
Decision was discussed in this thread: p1618406589275200-slack-CGGCLBN58

vbelolapotkov commented on client/stylesheets/abstracts/_colors.scss:
```diff
@@ -2,3 +2,12 @@
 @import 'node_modules/@automattic/color-studio/dist/color-variables.scss';
 @import 'node_modules/@wordpress/base-styles/colors.scss';
+
+// This is temp placeholders for missing variables after styles upgrade
+// Values taken from: https://github.com/WordPress/gutenberg/blob/%40wordpress/base-styles%401.0.0/packages/base-styles/_colors.scss
+// TODO: Replace variables with new color palette
+$light-gray-500: #e2e4e7;
+$light-gray-200: #f3f4f5;
+$light-gray-300: #edeff0;
+$dark-gray-900: #191e23;
+$dark-gray-500: #555d66;
```
@LevinMedia those color variables are removed from latest `@wordpress/base-styles`. Should we keep them as it is done now or replace with alternatives from `@automattic/color-studio`?

Links: [old base-styles (v1.0.0)](https://github.com/WordPress/gutenberg/blob/%40wordpress/base-styles%401.0.0/packages/base-styles/_colors.scss), [current base-styles (v3.4.1)](https://github.com/WordPress/gutenberg/blob/%40wordpress/base-styles%403.4.1/packages/base-styles/_colors.scss).

vbelolapotkov replied:
Replaced one by one in commits from https://github.com/Automattic/woocommerce-payments/pull/1536/commits/2f531cee48e97f2ddc840e75a89c969be5892086 to https://github.com/Automattic/woocommerce-payments/pull/1536/commits/b1fe1051def5f6075cee5a9ec2b1e522a795b888

waclawjacek commented:
I have just merged #1533 and this PR on a test branch and it works great! Styles for `<Button isSecondary />` seem to be spot on.

The only issue I noticed is that the styles for `<Button isLink />` are different than in the current `@wordpress/components` trunk. [Trunk uses:](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/button/style.scss#L213)

```css
color: var( --wp-admin-theme-color );
```

But [version 11.1.5](https://github.com/WordPress/gutenberg/blob/%40wordpress/components%4011.1.5/packages/components/src/button/style.scss#L198) used in this PR uses:

```css
color: #0073aa;
```

However, after a quick investigation [it turns out this was actually just changed](https://github.com/WordPress/gutenberg/pull/30823) in `trunk` and is not yet available in any of the releases, including the most recent 13.0.2.

jrodger reviewed (commented):
I've compared the UI from this branch with `develop`, the changes I noticed are below. I think they're all fine, just a query about the buttons on the ToS overlay.

<table>
<tr><th>New</th><th>Old</th><th>Notes</th></tr>
<tr>
<td>
<img src="https://user-images.githubusercontent.com/686419/115590641-1cc91b00-a2c9-11eb-9699-9afdf22dc15b.png"/>
</td>
<td>
<img src="https://user-images.githubusercontent.com/686419/115590840-5568f480-a2c9-11eb-8813-4f809c175d79.png"/>
</td>
<td>Updated button styles, looks good.</td>
</tr>
<tr>
<td>
<img src="https://user-images.githubusercontent.com/686419/115590929-74678680-a2c9-11eb-9a55-85582ec5e125.png"/>
</td>
<td>
<img src="https://user-images.githubusercontent.com/686419/115590963-80534880-a2c9-11eb-8e1f-f32ccfde3a88.png"/>
</td>
<td>Tweaked margins, looks better now IMO</td>
</tr>
<tr>
<td>
<img src="https://user-images.githubusercontent.com/686419/115591117-b0025080-a2c9-11eb-9e51-1114f212f089.png"/>
</td>
<td>
<img src="https://user-images.githubusercontent.com/686419/115591150-b690c800-a2c9-11eb-9ac4-9500ef7ae806.png"/>
</td>
<td>Updated and repositioned button</td>
</tr>
<tr>
<td>
<img src="https://user-images.githubusercontent.com/686419/115591430-ff488100-a2c9-11eb-821c-bba7afa93ee4.png"/>
</td>
<td>
<img src="https://user-images.githubusercontent.com/686419/115591462-05d6f880-a2ca-11eb-8224-ad5c7182cb41.png"/>
</td>
<td>The ToS modal buttons are the same, should this have new styles as well?</td>
</tr>
<tr>
<td>
<img src="https://user-images.githubusercontent.com/686419/115591531-15eed800-a2ca-11eb-82b3-a5b16cf1a82a.png"/>
</td>
<td>
<img src="https://user-images.githubusercontent.com/686419/115591555-1e471300-a2ca-11eb-8847-650e6e0a8d3f.png"/>
</td>
<td>The arrow has been dropped, but looks fine to me</td>
</tr>
</table>

This look ready to go, except for the couple of outstanding conversations above. On the webpack point, it doesn't sound like we have any option but to revert to v4? Was there anything you wanted input on for the base style colours?

LevinMedia commented:
@jrodger This is looking good! Thank you! 

For the evidence submission - can you please update it so that the save for later button is aligned left with the submit evidence button please?

For the TOS acceptance, we should update the decline button so that it's using the same default outline button

vbelolapotkov commented:
> @jrodger This is looking good! Thank you!
> 
> For the evidence submission - can you please update it so that the save for later button is aligned left with the submit evidence button please?

Fixed in https://github.com/Automattic/woocommerce-payments/pull/1536/commits/bf73863aa434172fcff5eae73a9c36de392e3612

> For the TOS acceptance, we should update the decline button so that it's using the same default outline button

Fixed in https://github.com/Automattic/woocommerce-payments/pull/1536/commits/3635ce37c0615c44c1680b04a9682a98212f9682

vbelolapotkov commented:
@jrodger I've addressed all updates requested by @LevinMedia and replaced deprecated colors using my best judgement which worth validating 😄 Anyway, it's in sync with develop and ready for final review.

jrodger reviewed (approved):
Retested, looks good!
